From 8cf66c4b5dd657ca14552b27a42901db0a5e9280 Mon Sep 17 00:00:00 2001
From: Herve Codina <herve.codina@bootlin.com>
Date: Mon, 4 Oct 2021 17:13:45 +0200
Subject: [PATCH] Fix compatibility with php8 - call_user_function_ex

This patch fixes compatibility with php8 replacing
call_user_function_ex by call_user_function.

Upstream: https://github.com/derickr/pecl-dbus/pull/8
Signed-off-by: Herve Codina <herve.codina@bootlin.com>
---
 dbus.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dbus.c b/dbus.c
index a98e2ea..6fa9897 100644
--- a/dbus.c
+++ b/dbus.c
@@ -1015,8 +1015,8 @@ php_dbus_do_method_call(php_dbus_obj *dbus,
 		method_args = safe_emalloc(sizeof(zval *), num_elems, 0);
 	}
 
-	if (call_user_function_ex(EG(function_table), object, &callback, &retval,
-				              num_elems, method_args, 0, NULL) == SUCCESS) {
+	if (call_user_function(EG(function_table), object, &callback, &retval,
+				              num_elems, method_args) == SUCCESS) {
 		if (!Z_ISUNDEF(retval)) {
 			reply = dbus_message_new_method_return(msg);
 			php_dbus_append_parameters(reply, &retval, NULL,
-- 
2.31.1

